manager/oracles.js

```diff
@@ -637,39 +637,40 @@ const instagram = async (UserId, link) => {
+                var resMediaAccessToken = (await rp.get(mediaGetNewAccessToken))
+                var media = `https://graph.facebook.com/${oauth.facebook.fbGraphVersion}/${instagram_id}/media?fields=like_count,shortcode,media_url&limit=50&access_token=${resMediaAccessToken.data.access_token}`
+                var resMedia = (await rp.get(media))
+                var data = resMedia.data.data
                 for (let i = 0; i < data.length; i++) {
                     if (data[i].shortcode == idPost) {
                         perf.likes = data[i].like_count
                         perf.media_url = data[i]?.media_url || ' '
-                        var mediaViews =
-                            'https://graph.facebook.com/' +
-                            oauth.facebook.fbGraphVersion +
-                            '/' +
-                            data[i].id +
-                            '/insights?metric=impressions&access_token=' +
-                            resMediaAccessToken.access_token
+                    // CHECK IF REEL 
+                    const mediaType = `https://graph.facebook.com/${oauth.facebook.fbGraphVersion}/${data[i].id}?fields=media_type,media_product_type&access_token=${resMediaAccessToken.data.access_token}`
+                    const mediaViews = `https://graph.facebook.com/${oauth.facebook.fbGraphVersion}/${data[i].id}/insights?metric=impressions,reach&access_token=${resMediaAccessToken.data.access_token}`
+                    const mediaViewReels = `https://graph.facebook.com/${oauth.facebook.fbGraphVersion}/${data[i].id}/insights?metric=plays&access_token=${resMediaAccessToken.data.access_token}`    
                         try {
-                            var resMediaViews = (await rp.get( mediaViews)).data
-                            let nbviews = JSON.stringify(resMediaViews)
-                            perf.views =
-                                JSON.parse(nbviews).data[0].values[0].value || 0
+
+                            // GET TYPE OF POST 
+                            var resMediaType = (await rp.get( mediaType))
+                            if(!!resMediaType.data) {
+                                if(resMediaType.data.media_type === 'VIDEO' && resMediaType.data.media_product_type === "REELS") {
+                                    var resMediaViews = (await rp.get( mediaViewReels))
+                                    perf.views = resMediaViews.data.data[0].values[0].value || 0
+                                } else {
+                                    var resMediaViews = (await rp.get( mediaViews))
+                                    perf.views = resMediaViews.data.data[0].values[0].value || 0
+                                }
+                            } else {
+                                var resMediaViews = (await rp.get( mediaViews))
+                                perf.views = resMediaViews.data.data[0].values[0].value || 0
+                            }
```

Comment:
The error handling in the code you provided could be improved in several ways. Here's a suggested change.

Currently, you have a `catch` statement at the end of your function that simply logs the error to the console. This can make debugging difficult because you don't know where the error occurred in your function. Moreover, the code that calls your function has no idea an error occurred since the error is not returned or re-thrown.

A better approach would be to re-throw the error after logging it, allowing the calling code to handle the error as it sees fit. Here's how you might modify your code:

```javascript
try {
    // your code here
} catch (err) {
    console.error('instagram manager oracle', err)
    throw err
}
```

With this change, the error is still logged to the console, but it is also re-thrown. This means that the code that calls your function can catch the error and handle it accordingly.

In addition, in the inner try/catch where you make a request to the Instagram API, you return a performance with views set to 0 in case of error. This is a design decision, but you might consider throwing a specific error at this level, which can give more details about the error that occurred. Here's how:

```javascript
try {
    // your code here
} catch (error) {
    console.error('Error when fetching Instagram data', error)
    throw new Error('Error when fetching Instagram data')
}
```

These changes will help to improve error handling in your code.